qa/tasks/deepsea.py

```diff
@@ -0,0 +1,98 @@
+'''
+Task to deploy clusters with DeepSea
+'''
+import logging
+
+from teuthology import misc
+from teuthology.orchestra import run
+from teuthology.salt import Salt
+from teuthology.task import Task
+from util import get_remote_for_role
+
+log = logging.getLogger(__name__)
+
+class DeepSea(Task):
+
+    def __init__(self, ctx, config):
+        super(DeepSea, self).__init__(ctx, config)
+
+        # determine the role id of the master role
+        if(misc.num_instances_of_type(self.cluster, 'master') != 1):
+            raise ConfigError('deepsea requires a single master role')
+        for id_ in misc.all_roles_of_type(self.ctx.cluster, 'master'):
+            # this loop should iterate only once
+            master_role = '.'.join(['master', str(id_)])
+
+        # set remote name for salt to pick it up. Setting the remote itself will
+        # crash the reporting tool since it doesn't know how to turn the object
+        # into a string
+
+        self.config["master_remote"] = get_remote_for_role(self.ctx,
+                master_role).name
+        self.salt = Salt(self.ctx, self.config)
+
+    def setup(self):
+        super(DeepSea, self).setup()
+
+        self.log.info("master remote: {}".format(self.config["master_remote"]))
+
+        self.salt.master_remote.run(args=[
+            'git',
+            'clone',
+            '--branch',
+            'wip-teuthology-suite',
+            'https://github.com/smithfarm/DeepSea.git',
```

Comment:
This needs to be rewritten so the `deepsea` task takes `repo` and `branch` options that default to `https://github.com/SUSE/DeepSea` and `master`, respectively.